### PR TITLE
Bump ecmarkdown to 4.0.0

### PR DESCRIPTION
**Pull request by bakkot**
This should fail until I push an additional commit cleaning up the errors the linter can now identify. It will also need https://github.com/tc39/ecma262/pull/2133 to be merged.

### CODE REVIEW

File: spec.html [hunk 1]
```diff
@@ -27605,7 +27605,7 @@ <h1>Properties of the Number Constructor</h1>
 
       <emu-clause id="sec-number.epsilon">
         <h1>Number.EPSILON</h1>
-        <p>The value of `Number.EPSILON` is the difference between 1 and the smallest value greater than 1 that is representable as a Number value, which is approximately 2.2204460492503130808472633361816 x 10<sup> - 16</sup>.</p>
+        <p>The value of `Number.EPSILON` is the difference between 1 and the smallest value greater than 1 that is representable as a Number value, which is approximately 2.2204460492503130808472633361816 &times; 10<sup>-16</sup>.</p>
```
Comment (bakkot) [inline spec.html:27608]: The ` - 16` -> `-16` change affects the rendering, but I consider it to be an improvement. It also makes this match what we do in Number.MIN_VALUE, which we describe as being approximately `5 &times; 10<sup>-324</sup>`.

Comment (ljharb) [reply spec.html:27608]: I very much agree this is an improvement
https://tc39.es/ecma262/#sec-number.epsilon vs https://ci.tc39.es/preview/tc39/ecma262/sha/285329afcf5011293bbf5b85f72beff1e8b54d21/#sec-number.epsilon

File: spec.html [hunk 2]
```diff
@@ -112,7 +112,7 @@ <h1>Introduction</h1>
...
+  <p>ECMAScript 2020, the 11<sup>th</sup> edition, introduces the `matchAll` method for Strings, to produce an iterator for all match objects generated by a global regular expression; `import()`, a syntax to asynchronously import Modules with a dynamic specifier; `BigInt`, a new number primitive for working with arbitrary precision integers; `Promise.allSettled`, a new Promise combinator that does not short-circuit; `globalThis`, a universal way to access the global `this` value; dedicated `export * as ns from 'module'` syntax for use within modules; increased standardization of `for-in` enumeration order; `import.meta`, a host-populated object available in Modules that may contain contextual information about the Module; as well as adding two new syntax features to improve working with “nullish” values (`null` or `undefined`): nullish coalescing, a value selection operator; and optional chaining, a property access and function invocation operator that short-circuits if the value to access/invoke is nullish.</p>
```
Comment (ljharb) [inline spec.html:115]: the diff on github doesn't show it, but there's an extra space removed after `about the Module;`.

File: spec.html [hunk 3]
```diff
@@ -4566,7 +4566,7 @@ <h1>The Abstract Closure Specification Type</h1>
...
+      <p>A data block that resides in memory that can be referenced from multiple agents concurrently is designated a <dfn>Shared Data Block</dfn>. A Shared Data Block has an identity (for the purposes of equality testing Shared Data Block values) that is <em>address-free</em>: it is tied not to the virtual addresses the block is mapped to in any process, but to the set of locations in memory that the block represents. Two data blocks are equal only if the sets of the locations they contain are equal; otherwise, they are not equal and the intersection of the sets of locations they contain is empty. Finally, Shared Data Blocks can be distinguished from Data Blocks.</p>
```
Comment (ljharb) [inline spec.html:4569]: removed extra space after `they contain is empty.`

File: spec.html [hunk 4]
```diff
@@ -8043,15 +8043,15 @@ <h1>Agent Clusters</h1>
 
     <p>All agents within a cluster must have different values for the [[Signifier]] property in their respective Agent Records.</p>
 
-    <p>An embedding may deactivate (stop forward progress) or activate (resume forward progress) an agent without the agent's knowledge or cooperation.  If the embedding does so, it must not leave some agents in the cluster active while other agents in the cluster are deactivated indefinitely.</p>
+    <p>An embedding may deactivate (stop forward progress) or activate (resume forward progress) an agent without the agent's knowledge or cooperation. If the embedding does so, it must not leave some agents in the cluster active while other agents in the cluster are deactivated indefinitely.</p>
```
Comment (ljharb) [inline spec.html:8046]: removed extra space before "If the embedding"

File: spec.html [hunk 5]
```diff
@@ -8043,15 +8043,15 @@ <h1>Agent Clusters</h1>
...
+      <p>The purpose of the preceding restriction is to avoid a situation where an agent deadlocks or starves because another agent has been deactivated. For example, if an HTML shared worker that has a lifetime independent of documents in any windows were allowed to share memory with the dedicated worker of such an independent document, and the document and its dedicated worker were to be deactivated while the dedicated worker holds a lock (say, the document is pushed into its window's history), and the shared worker then tries to acquire the lock, then the shared worker will be blocked until the dedicated worker is activated again, if ever. Meanwhile other workers trying to access the shared worker from other windows will starve.</p>
```
Comment (ljharb) [inline spec.html:8049]: removed extra space after "has been deactivated"

File: spec.html [hunk 6]
```diff
@@ -8043,15 +8043,15 @@ <h1>Agent Clusters</h1>
...
+    <p>An embedding may terminate an agent without any of the agent's cluster's other agents' prior knowledge or cooperation. If an agent is terminated not by programmatic action of its own or of another agent in the cluster but by forces external to the cluster, then the embedding must choose one of two strategies: Either terminate all the agents in the cluster, or provide reliable APIs that allow the agents in the cluster to coordinate so that at least one remaining member of the cluster will be able to detect the termination, with the termination data containing enough information to identify the agent that was terminated.</p>
```
Comment (ljharb) [inline spec.html:8054]: removed before "If an agent"

File: spec.html [hunk 7]
```diff
@@ -8071,7 +8071,7 @@ <h1>Agent Clusters</h1>
   <emu-clause id="sec-forward-progress">
     <h1>Forward Progress</h1>
     <p>For an agent to <em>make forward progress</em> is for it to perform an evaluation step according to this specification.</p>
-    <p>An agent becomes <em>blocked</em> when its running execution context waits synchronously and indefinitely for an external event. Only agents whose Agent Record's [[CanBlock]] property is *true* can become blocked in this sense.  An <em>unblocked</em> agent is one that is not blocked.</p>
+    <p>An agent becomes <em>blocked</em> when its running execution context waits synchronously and indefinitely for an external event. Only agents whose Agent Record's [[CanBlock]] property is *true* can become blocked in this sense. An <em>unblocked</em> agent is one that is not blocked.</p>
```
Comment (ljharb) [inline spec.html:8074]: removed after "in this sense."

**Pull request merged**